kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationIT.java

```diff
@@ -0,0 +1,282 @@
+
+            var saslHandshakeRequestsGoingUpCount = findFirstMetricMatching(allMetrics, UPSTREAM_SASL_HANDSHAKE_LABELS_PREDICATE);
+            var saslAuthenticationResponsesComingDown = findFirstMetricMatching(allMetrics, DOWNSTREAM_SASL_AUTHENTICATE_PREDICATE);
+
+            assertThat(saslHandshakeRequestsGoingUpCount)
+                    .isPresent()
+                    .get(DOUBLE)
+                    .withFailMessage("Expecting proxy to have seen at least one handshake requests from downstream")
+                    .isPositive();
+
+            assertThat(saslAuthenticationResponsesComingDown).isNotPresent();
+        }
+        finally {
+            @SuppressWarnings("java:S1481")
+            var ignored = badTokenFile.toFile().delete();
+        }
+
+    }
+
+    private Optional<Double> findFirstMetricMatching(List<SimpleMetric> all, Predicate<SimpleMetric> predicate) {
+        return all.stream()
+                .filter(predicate)
+                .findFirst()
+                .map(SimpleMetric::value);
+    }
+
+    private ConfigurationBuilder getConfiguredProxyBuilder() {
+        return proxy(cluster)
+                .withNewAdminHttp()
+                .withNewEndpoints()
+                .withNewPrometheus()
+                .endPrometheus()
+                .endEndpoints()
+                .endAdminHttp()
+                .addToFilters(new FilterDefinitionBuilder(
+                        OauthBearerValidation.class.getName())
+                        .withConfig("jwksEndpointUrl", JWKS_ENDPOINT_URL,
+                                "expectedAudience", EXPECTED_AUDIENCE)
+                        .build());
+    }
+
+    @NonNull
+    private Map<String, Object> getClientConfig(URI tokenEndpointUrl) {
+        var oauthClientConfig = new HashMap<String, Object>();
+        oauthClientConfig.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SASL_PLAINTEXT.name());
+        oauthClientConfig.put(SaslConfigs.SASL_MECHANISM, OAuthBearerLoginModule.OAUTHBEARER_MECHANISM);
+        oauthClientConfig.put(SaslConfigs.SASL_OAUTHBEARER_TOKEN_ENDPOINT_URL, tokenEndpointUrl.toString());
+        oauthClientConfig.put(SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS, OAuthBearerLoginCallbackHandler.class.getCanonicalName());
+        oauthClientConfig.put(SaslConfigs.SASL_JAAS_CONFIG,
+                "%s required clientId=\"clientIdIgnore\" clientSecret=\"clientSecretIgnore\";".formatted(OAuthBearerLoginModule.class.getName()));
+        return oauthClientConfig;
+    }
+
+    /**
+     * Pings the cluster in order to assert connectivity. We don't care about the result.
+     * @param admin admin
+     */
+    @SuppressWarnings("java:S1481") // making clear the intent that the result of the operation is unneeded.
+    private void performClusterOperation(Admin admin) {
+        var unused = admin.describeCluster().nodes().toCompletionStage().toCompletableFuture().join();
+        assertThat(unused).isNotNull();
```

Comment:
Would 
```suggestion
assertThat(admin.describeCluster().nodes().toCompletionStage().toCompletableFuture()).succeedsWithin(10, TimeUnit.SECONDS);
```

be clearer? I'm not sure what the default junit timeout is (if there even is one) but this could in theory block for a long time...

Reply:
I couldn't do as you suggested, as authWithBadToken needs to see the SaslAuthenticationException.  But I refactored to avoid the possibility of an long wait.

Reply:
I think you could have used `failsWithin(xxx).withCauseInstanceOf(SaslAuthenticationException.class)` but anyway thats not a big deal.